acra-censor/handlers/querycapture_handler.go

```diff
@@ -152,18 +111,15 @@ func (handler *QueryCaptureHandler) CheckQuery(query string) (bool, error) {
 	queryInfo.RawQuery = query
 	queryInfo.IsForbidden = false
 	handler.Queries = append(handler.Queries, *queryInfo)
-
-	select {
-	case handler.logChannel <- *queryInfo: // channel is ok
-	default: //channel is full
-		log.Errorf("Can't process too many queries. Skip query: %s", trimToN(query, LogQueryLength))
-	}
+	handler.BufferedQueries = append(handler.BufferedQueries, *queryInfo)
```

Comment:
better to append pointers instead copy of struct. change type of BufferedQueries from `QueryInfo` to `*QueryInfo` and then append only pointers.